smithy-aws-protocol-tests/model/restJson1/http-response-code.smithy

```diff
@@ -73,3 +93,31 @@ apply HttpResponseCode @httpResponseTests([
         appliesTo: "client"
     },
 ])
+
+apply ResponseCodeRequired @httpResponseTests([
+    {
+        id: "RestJsonHttpResponseCodeRequired",
+        documentation: """
+                This test ensures that servers handle @httpResponseCode being @required.""",
+        protocol: restJson1,
+        code: 201,
+        body: "",
```

Comment:
See similar tests https://github.com/awslabs/smithy/blob/ededf6ba2334f1d006f3c76a257c8812f2f17497/smithy-aws-protocol-tests/model/restJson1/empty-input-output.smithy#L147-L159 and https://github.com/awslabs/smithy/blob/ededf6ba2334f1d006f3c76a257c8812f2f17497/smithy-aws-protocol-tests/model/restJson1/empty-input-output.smithy#L219-L221

> server implementations are expected to respond with a JSON object regardless of if the output parameters are empty

I think they way these tests are defined, it would have failed in the TypeScript SSDK - [example](https://github.com/gosar/aws-sdk-js-v3/blob/directed-codegen/private/aws-restjson-server/test/functional/restjson1.spec.ts#L27470-L27523)

Reply:
Thanks @gosar, updated

Reply:
> I think they way these tests are defined, it would have failed in the TypeScript SSDK - [example](https://github.com/gosar/aws-sdk-js-v3/blob/directed-codegen/private/aws-restjson-server/test/functional/restjson1.spec.ts#L27470-L27523)

Would it be useful to add some tests from those in this CI?